src/engine/ScreenFilter.cs

```diff
@@ -0,0 +1,37 @@
+using Godot;
+
+public class ScreenFilter : TextureRect
+{
+    private TextureRect screenFilter;
+    private ShaderMaterial screenFilterMaterial;
+    public override void _Ready()
+    {
+        screenFilter = (TextureRect)this;
```

Comment:
I find this pretty pointless. All of the places that use `screenFilter.something` would work just as well being `something` right?